jetty-websocket/jetty-websocket-client/src/main/java/org/eclipse/jetty/websocket/client/WebSocketClient.java

```diff
@@ -61,23 +63,17 @@
     private final JettyWebSocketFrameHandlerFactory frameHandlerFactory;
     private final List<WebSocketSessionListener> sessionListeners = new CopyOnWriteArrayList<>();
     private final SessionTracker sessionTracker = new SessionTracker();
-    private ClassLoader contextClassLoader;
+    private final FrameHandler.ConfigurationCustomizer configurationCustomizer = new FrameHandler.ConfigurationCustomizer();
     private DecoratedObjectFactory objectFactory;
     private WebSocketExtensionRegistry extensionRegistry;
-    private int inputBufferSize = 4 * 1024;
-    private int outputBufferSize = 4 * 1024;
-    private long maxBinaryMessageSize = 64 * 1024;
-    private long maxTextMessageSize = 64 * 1024;
 
     /**
      * Instantiate a WebSocketClient with defaults
      */
     public WebSocketClient()
     {
-        this(new WebSocketCoreClient());
-        this.coreClient.getHttpClient().setName("Jetty-WebSocketClient@" + hashCode());
-        // We created WebSocketCoreClient, let lifecycle be managed by us
-        addManaged(coreClient);
+        this(null);
+        coreClient.getHttpClient().setName("Jetty-WebSocketClient@" + hashCode());
```

Comment:
This should be in the main constructor and done after creating the coreClient if the passed in httpClient is null